dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/batching/MigratorFactory.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.batching
+
+import com.datadog.android.core.internal.data.batching.migrators.BatchedDataMigrator
+import com.datadog.android.core.internal.data.privacy.Consent
+
+internal class MigratorFactory(
```

Comment:
```kotlin
interface MigratorFactory {
    fun resolveMigrator(
        prevConsentFlag: Consent?,
        newConsentFlag: Consent
    ): BatchedDataMigrator
}
```

```kotlin
class FolderMigratorFactory(
    val pendingFolderPath:String,
    val approvedFolderPath:String
) : MigratorFactory {
    override fun resolveMigrator(
        prevConsentFlag: Consent?,
        newConsentFlag: Consent
    ): BatchedDataMigrator {
        if (prevConsentFlag == null) return NoOp()
       
        val fromFolder = folderFromConsent(prevConsentFlag)
        val destFolder = folderFromConsent(newConsentFlag)
        return FolderMigrator(fromFolder, destFolder)
    }
}
```

Reply:
I actually think we need something like :
```
class MoveDataMigrator(pendingFolderPath, approvedFolderPath)
class WipeDataMigrator(pendingFolderPath)
class NoOpMigrator()
```